optimum/habana/diffusers/pipelines/cogvideox/pipeline_cogvideox_gaudi.py

```diff
@@ -0,0 +1,846 @@
+                end_frame = frame_batch_size * (k + 1) + remaining_frames
+                tile = z[
+                    :,
+                    :,
+                    start_frame:end_frame,
+                    i : i + self.tile_latent_min_height,
+                    j : j + self.tile_latent_min_width,
+                ].clone()
+                if self.post_quant_conv is not None:
+                    tile = self.post_quant_conv(tile)
+                tile, conv_cache = self.decoder(tile, conv_cache=conv_cache)
+                time.append(tile.clone())
+                htcore.mark_step()
+
+            row.append(torch.cat(time, dim=2))
+        rows.append(row)
+
+    result_rows = []
+    for i, row in enumerate(rows):
+        result_row = []
+        for j, tile in enumerate(row):
+            # blend the above tile and the left tile
+            # to the current tile and add the current tile to the result row
+            if i > 0:
+                tile = self.blend_v(rows[i - 1][j], tile, blend_extent_height)
+            if j > 0:
+                tile = self.blend_h(row[j - 1], tile, blend_extent_width)
+            result_row.append(tile[:, :, :, :row_limit_height, :row_limit_width])
+        result_rows.append(torch.cat(result_row, dim=4))
+
+    dec = torch.cat(result_rows, dim=3)
+
+    if not return_dict:
+        return (dec,)
+
+    return DecoderOutput(sample=dec)
+
+
+def CogVideoXCausalConv3dforwardGaudi(
+    self, inputs: torch.Tensor, conv_cache: Optional[torch.Tensor] = None
+) -> torch.Tensor:
+    # print('run gaudi CogVideoXCausalConv3d forward!')
+    inputs = self.fake_context_parallel_forward(inputs, conv_cache)
+    # conv_cache = inputs[:, :, -self.time_kernel_size + 1 :].clone()
+
+    padding_2d = (self.width_pad, self.width_pad, self.height_pad, self.height_pad)
+    inputs_pad = F.pad(inputs, padding_2d, mode="constant", value=0)
+
+    output = self.conv(inputs_pad)
+    if self.time_kernel_size > 1:
+        if conv_cache is not None and conv_cache.shape == inputs[:, :, -self.time_kernel_size + 1 :].shape:
+            conv_cache.copy_(inputs[:, :, -self.time_kernel_size + 1 :])
+        else:
+            conv_cache = inputs[:, :, -self.time_kernel_size + 1 :].clone()
+    return output, conv_cache
+
+
+setattr(CogVideoXCausalConv3d, "forward", CogVideoXCausalConv3dforwardGaudi)
+setattr(AutoencoderKLCogVideoX, "tiled_decode", tiled_decode_gaudi)
+
+
+@dataclass
+class GaudiTextToVideoSDPipelineOutput(BaseOutput):
+    r"""
+    Output class for CogVideo pipelines.
+
+    Args:
+        frames (`torch.Tensor`, `np.ndarray`, or List[List[PIL.Image.Image]]):
+            List of video outputs - It can be a nested list of length `batch_size,` with each sub-list containing
+            denoised PIL image sequences of length `num_frames.` It can also be a NumPy array or Torch tensor of shape
+            `(batch_size, num_frames, channels, height, width)`.
+    """
+
+    frames: torch.Tensor
```

Comment:
It's exactly the same as https://github.com/huggingface/diffusers/blob/4397f59a37012054212603a8dc8ba9db58f8a56c/src/diffusers/pipelines/cogvideo/pipeline_output.py#L9, let's import it directly here instead of duplicating code